recipes/orc/all/patches/0001-thirdparty-toolchain.patch

```diff
@@ -0,0 +1,55 @@
+diff --git a/cmake_modules/ConanThirdpartyToolchain.cmake b/cmake_modules/ConanThirdpartyToolchain.cmake
+new file mode 100644
+index 000000000..1df89e849
+--- /dev/null
++++ b/cmake_modules/ConanThirdpartyToolchain.cmake
+@@ -0,0 +1,49 @@
++# ----------------------------------------------------------------------
++# Snappy
++
++find_package (Snappy REQUIRED)
++
++add_library (orc_snappy INTERFACE)
++add_library (orc::snappy ALIAS orc_snappy)
++target_link_libraries(orc_snappy INTERFACE ${Snappy_LIBRARIES})
++target_include_directories (orc_snappy SYSTEM INTERFACE ${Snappy_INCLUDE_DIR})
++
++# ----------------------------------------------------------------------
++# ZLIB
++
++find_package (ZLIB REQUIRED)
++
++add_library (orc_zlib INTERFACE)
++add_library (orc::zlib ALIAS orc_zlib)
++target_link_libraries (orc_zlib INTERFACE ${ZLIB_LIBRARIES})
++target_include_directories (orc_zlib SYSTEM INTERFACE ${ZLIB_INCLUDE_DIR})
```

Comment:
You can probably avoid most of these steps in this .cmake file by renaming the generated CMake targets in `generate()`:
```python
        deps = CMakeDeps(self)
        deps.set_property("zlib", "cmake_target_name", "orc::zlib")
        ...
        deps.generate()
```

Which is basically what I did in my PR: https://github.com/conan-io/conan-center-index/pull/21007/files#diff-3cccf7963994e6634e694bd42cc4cf63970514d07a4cdf290d2f90ab0aefcb0bR118-R121

Reply:
Thanks for the suggestion! I'd go with https://github.com/conan-io/conan-center-index/pull/22854#discussion_r1501853311 as we have to customize some logic anyway in the cmake file and this is not required any more after ORC v2.0.0.